app/Http/ViewComposers/MenuComposer.php

```diff
@@ -223,6 +223,10 @@ public function compose(View $view)
             ->isActive()
             ->hasAccess($user)
             ->groupBy('severity')
+            ->leftJoin('devices', 'alerts.device_id', '=', 'devices.device_id')
+            ->where('devices.disabled', '=', '0')
+            ->where('devices.ignore', '=', '0')
```

Comment:
Should we include this in "isActive" ? I mean, an alert can be active if and only if those 2 where conditions on the device are met.

Reply:
Not in a simple way.  Joining tables can have implications on the query, so I don't like hiding it in a scope.

Reply:
But maybe :D

Reply:
I'll follow your 1st advice and keep it as it is for now.